### PR TITLE
fix(gents): PR2/2 Fix some gents tests so output can compile.

### PR DESCRIPTION
**Pull request by bowenni**
Prepare for https://github.com/angular/clutz/issues/404

### CODE REVIEW

File: src/test/java/com/google/javascript/gents/singleTests/externs_map.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+declare namespace ng {
```
Comment (bowenni) [inline src/test/java/com/google/javascript/gents/singleTests/externs_map.d.ts]: Used for compilation.

File: src/test/java/com/google/javascript/gents/singleTests/fields.js [hunk 2]
```diff
@@ -35,9 +36,9 @@ A.prototype.foo = function() {
   this.d;
 
   this.n = 14;
+  this.x = this.a;
   // These are undeclared fields
   this.u;
-  this.x = this.a;
```
Comment (bowenni) [inline src/test/java/com/google/javascript/gents/singleTests/fields.js]: Closure WARNING Property 'a' is undefined.

File: src/test/java/com/google/javascript/gents/singleTests/goog_provide.js [hunk 3]
```diff
@@ -32,4 +32,3 @@ A.B.C = function() {};
 A.B.C.bar = function() { return false; };
 
 A.B.D = path.to.someUtilFunction();
-A.B.D.setA(1).setB(2);
```
Comment (bowenni) [inline src/test/java/com/google/javascript/gents/singleTests/goog_provide.js]: Closure WARNING - No properties on this expression

File: src/test/java/com/google/javascript/gents/singleTests/interface.js [hunk 4]
```diff
@@ -27,14 +27,30 @@ ns.Interface3 = function() {}
 ns.Interface3.prototype.baz = function(a) {};
 
 /** @implements {ns.Interface2} */
-class X {}
```
Comment (bowenni) [inline src/test/java/com/google/javascript/gents/singleTests/interface.js]: Correctly implements interface2.

File: src/test/java/com/google/javascript/gents/singleTests/undeclared_property.d.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+declare namespace goog {
```
Comment (bowenni) [inline src/test/java/com/google/javascript/gents/singleTests/undeclared_property.d.ts]: Used for compilation.

File: src/test/java/com/google/javascript/gents/singleTests/externs_map.d.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+  interface IDirective {
+  }
+}
+
+declare type JQuery = {};
+declare type OldNoType = {};
```
Comment (gmoothart) [inline src/test/java/com/google/javascript/gents/singleTests/externs_map.d.ts]: this is meant to be `any`, but I guess this is ok if it compiles

File: src/test/java/com/google/javascript/gents/singleTests/externs_map.d.ts (hunk 1 above)
Comment (mprobst) [inline src/test/java/com/google/javascript/gents/singleTests/externs_map.d.ts]: Add a comment `/** @fileoverview */` explaining that?

File: src/test/java/com/google/javascript/gents/singleTests/undeclared_property.d.ts (hunk 5 above)
Comment (mprobst) [inline src/test/java/com/google/javascript/gents/singleTests/undeclared_property.d.ts]: As above, please add a comment.

Comment (bowenni) [inline src/test/java/com/google/javascript/gents/singleTests/undeclared_property.d.ts]: Done.

File: src/test/java/com/google/javascript/gents/singleTests/externs_map.d.ts (hunk 1 above)
Comment (bowenni) [inline src/test/java/com/google/javascript/gents/singleTests/externs_map.d.ts]: Done.

**Pull request merged**